<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1108</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>971</width>
      <height>531</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="tabRF">
     <attribute name="title">
      <string>Root Finders</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>921</width>
        <height>442</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <layout class="QFormLayout" name="formLayout_2">
         <item row="1" column="0" colspan="2">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>Input:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_10">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Input a &lt;span style=&quot; text-decoration: underline;&quot;&gt;single variable&lt;/span&gt; function in terms of &lt;span style=&quot; font-weight:600;&quot;&gt;x&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Function</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="funcInpRoot">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Input a &lt;span style=&quot; text-decoration: underline;&quot;&gt;single variable&lt;/span&gt; function in terms of &lt;span style=&quot; font-weight:600;&quot;&gt;x&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>Precision</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="precisionRF">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="placeholderText">
            <string>1.0e-9</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_22">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;These will be the &lt;span style=&quot; font-weight:600;&quot;&gt;approximations&lt;/span&gt; to the root for the &lt;span style=&quot; text-decoration: underline;&quot;&gt;Secant&lt;/span&gt; and &lt;span style=&quot; text-decoration: underline;&quot;&gt;Reular Falsi&lt;/span&gt; methods.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Lower Bound</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QDoubleSpinBox" name="lowerBound">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;These will be the &lt;span style=&quot; font-weight:600;&quot;&gt;approximations&lt;/span&gt; to the root for the &lt;span style=&quot; text-decoration: underline;&quot;&gt;Secant&lt;/span&gt; and &lt;span style=&quot; text-decoration: underline;&quot;&gt;Reular Falsi&lt;/span&gt; methods.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-99999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>99999.990000000005239</double>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_23">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;These will be the &lt;span style=&quot; font-weight:600;&quot;&gt;approximations&lt;/span&gt; to the root for the &lt;span style=&quot; text-decoration: underline;&quot;&gt;Secant&lt;/span&gt; and &lt;span style=&quot; text-decoration: underline;&quot;&gt;Reular Falsi&lt;/span&gt; methods.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Upper Bound</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QDoubleSpinBox" name="upperBound">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;These will be the &lt;span style=&quot; font-weight:600;&quot;&gt;approximations&lt;/span&gt; to the root for the &lt;span style=&quot; text-decoration: underline;&quot;&gt;Secant&lt;/span&gt; and &lt;span style=&quot; text-decoration: underline;&quot;&gt;Reular Falsi&lt;/span&gt; methods.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-99999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>99999.990000000005239</double>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>Method</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QRadioButton" name="inpBi">
           <property name="text">
            <string>Bisection</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">btnGrpRF</string>
           </attribute>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QRadioButton" name="inpNR">
           <property name="text">
            <string>Newton Raphson</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">btnGrpRF</string>
           </attribute>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QRadioButton" name="inpRegF">
           <property name="text">
            <string>Regula Falsi</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">btnGrpRF</string>
           </attribute>
          </widget>
         </item>
         <item row="11" column="1">
          <widget class="QRadioButton" name="inpSec">
           <property name="text">
            <string>Secant</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">btnGrpRF</string>
           </attribute>
          </widget>
         </item>
         <item row="14" column="0">
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="15" column="0" colspan="2">
          <widget class="QPushButton" name="btnCalcRoot">
           <property name="text">
            <string>Calculate</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Initial Root Guess</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="guessX">
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-99999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>99999.990000000005239</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="2">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Output:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTextBrowser" name="outTextRoot"/>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tabLA">
     <attribute name="title">
      <string>Linear Algebra Solvers</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>931</width>
        <height>431</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="grdLAS">
       <item row="9" column="1">
        <widget class="QRadioButton" name="inpGS">
         <property name="text">
          <string>Gauss Seidel (Iterative)</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">btnGrpLAS</string>
         </attribute>
        </widget>
       </item>
       <item row="1" column="2">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Method</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPlainTextEdit" name="inpTextLA">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This &lt;span style=&quot; font-weight:600;&quot;&gt;will &lt;/span&gt;&lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;FAIL&lt;/span&gt; if the matrix is not &lt;span style=&quot; font-weight:600;&quot;&gt;SQUARE&lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Input (Ax=b):</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <widget class="QRadioButton" name="inpGJ">
         <property name="text">
          <string>Gauss Jordan</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">btnGrpLAS</string>
         </attribute>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>Vector b</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLineEdit" name="inpVecB">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Input a &lt;span style=&quot; text-decoration: underline;&quot;&gt;single variable&lt;/span&gt; function in terms of &lt;span style=&quot; font-weight:600;&quot;&gt;x&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="placeholderText">
          <string/>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QRadioButton" name="inpGE">
         <property name="text">
          <string>Gauss Elimination (Partial Pivoting)</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">btnGrpLAS</string>
         </attribute>
        </widget>
       </item>
       <item row="10" column="0">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_20">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This &lt;span style=&quot; font-weight:600;&quot;&gt;will &lt;/span&gt;&lt;span style=&quot; font-weight:600; text-decoration: underline;&quot;&gt;FAIL&lt;/span&gt; if the matrix is not &lt;span style=&quot; font-weight:600;&quot;&gt;SQUARE&lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Matrix A</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Output:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_24">
         <property name="text">
          <string>Maximum Iterations</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QLineEdit" name="inpLASMaxItr">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Input a &lt;span style=&quot; text-decoration: underline;&quot;&gt;single variable&lt;/span&gt; function in terms of &lt;span style=&quot; font-weight:600;&quot;&gt;x&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="placeholderText">
          <string>1000</string>
         </property>
        </widget>
       </item>
       <item row="11" column="0" colspan="2">
        <widget class="QPushButton" name="btnCalcLAS">
         <property name="text">
          <string>Calculate</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4" rowspan="11">
        <widget class="QTextBrowser" name="outTextLA"/>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tabODE">
     <attribute name="title">
      <string>ODE Solvers</string>
     </attribute>
     <widget class="QWidget" name="">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>931</width>
        <height>471</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="3" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Start x</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Method</string>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <spacer name="verticalSpacer_6">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="6" column="2">
        <widget class="QDoubleSpinBox" name="stepSize">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Defaults to &lt;span style=&quot; font-weight:600;&quot;&gt;0.025&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="decimals">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_6">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Defaults to &lt;span style=&quot; font-weight:600;&quot;&gt;0.025&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Step size</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Start y</string>
         </property>
        </widget>
       </item>
       <item row="2" column="5" rowspan="12">
        <widget class="QTextBrowser" name="outTextODE"/>
       </item>
       <item row="10" column="2">
        <widget class="QRadioButton" name="inpRK4">
         <property name="text">
          <string>Runge Kutta Fourth Order</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>End x</string>
         </property>
        </widget>
       </item>
       <item row="13" column="0" colspan="3">
        <widget class="QPushButton" name="btnCalcODE">
         <property name="text">
          <string>Calculate</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Function</string>
         </property>
        </widget>
       </item>
       <item row="9" column="2">
        <widget class="QRadioButton" name="inpRK3">
         <property name="text">
          <string>Runge Kutta Third Order</string>
         </property>
        </widget>
       </item>
       <item row="7" column="2">
        <widget class="QRadioButton" name="inpEu">
         <property name="text">
          <string>Euler (Runge Kutta First Order)</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QDoubleSpinBox" name="startX">
         <property name="decimals">
          <number>10</number>
         </property>
         <property name="minimum">
          <double>-99999999999.000000000000000</double>
         </property>
         <property name="maximum">
          <double>99999999999.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="5" rowspan="2">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Output:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="8" column="2">
        <widget class="QRadioButton" name="inpRK2">
         <property name="text">
          <string>Runge Kutta Second Order</string>
         </property>
        </widget>
       </item>
       <item row="11" column="2">
        <widget class="QRadioButton" name="inpMP">
         <property name="text">
          <string>Milne's Predictor Corrector</string>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QDoubleSpinBox" name="endX">
         <property name="decimals">
          <number>10</number>
         </property>
         <property name="minimum">
          <double>-99999999999.000000000000000</double>
         </property>
         <property name="maximum">
          <double>99999999999.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLineEdit" name="funcInpODE"/>
       </item>
       <item row="4" column="2">
        <widget class="QDoubleSpinBox" name="startY">
         <property name="decimals">
          <number>10</number>
         </property>
         <property name="minimum">
          <double>-99999999999.000000000000000</double>
         </property>
         <property name="maximum">
          <double>99999999999.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="0" rowspan="2" colspan="3">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Input:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="12" column="0">
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="btnQuit">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>570</y>
      <width>191</width>
      <height>32</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;I will &lt;span style=&quot; font-weight:600;&quot;&gt;end&lt;/span&gt; your suffering.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>Quit</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1108</width>
     <height>31</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="btnGrpRF"/>
  <buttongroup name="btnGrpLAS"/>
 </buttongroups>
</ui>
